contracts/LiquidationsPeripheral.vy

```diff
@@ -598,70 +598,72 @@ def setCryptoPunksAddress(_address: address):
 
 @external
 def addLiquidation(
-    _collateralAddress: address,
-    _tokenId: uint256,
     _borrower: address,
     _loanId: uint256,
     _erc20TokenContract: address
-):
-    assert ICollateralVaultPeripheral(self.collateralVaultPeripheralAddress).isCollateralInVault(_collateralAddress, _tokenId), "collateral not owned by vault"
-    
+):  
     borrowerLoan: Loan = ILoansCore(self.loansCoreAddresses[_erc20TokenContract]).getLoan(_borrower, _loanId)
     assert borrowerLoan.defaulted, "loan is not defaulted"
-    assert self._isCollateralInArray(borrowerLoan.collaterals, _collateralAddress, _tokenId), "collateral not in loan"
+    assert not ILiquidationsCore(self.liquidationsCoreAddress).isLoanLiquidated(_borrower, self.loansCoreAddresses[_erc20TokenContract], _loanId), "loan already liquidated"
```

Comment:
If the `LoansCore` contract  gets updated, wouldn't result in `isLoanLiquidated` returning `False` for the old loans?

Reply:
It would, yes, but if the `LoansCore` gets updated, it would mean a protocol upgrade since we can't migrate the loans stored there. This means that the new `LoansCore` would have new loans only. Does it make sense?